yt_dlp/utils.py

```diff
@@ -410,54 +410,126 @@ def xpath_attr(node, xpath, key, name=None, fatal=False, default=NO_DEFAULT):
     return n.attrib[key]
 
 
-def get_element_by_id(id, html):
-    """Return the content of the tag with the specified ID in the passed HTML document"""
-    return get_element_by_attribute('id', id, html)
+def get_element_by_id(id, html, return_content=True):
+    """Return the content or the tag with the specified ID in the passed HTML document"""
+    return get_element_by_attribute('id', id, html, return_content=return_content)
 
 
-def get_element_by_class(class_name, html):
-    """Return the content of the first tag with the specified class in the passed HTML document"""
-    retval = get_elements_by_class(class_name, html)
+def get_element_by_class(class_name, html, return_content=True):
+    """Return the content or the first tag with the specified class in the passed HTML document"""
+    retval = get_elements_by_class(class_name, html, return_content)
     return retval[0] if retval else None
 
 
-def get_element_by_attribute(attribute, value, html, escape_value=True):
-    retval = get_elements_by_attribute(attribute, value, html, escape_value)
+def get_element_by_attribute(attribute, value, html, escape_value=True, return_content=True):
+    retval = get_elements_by_attribute(attribute, value, html, escape_value, return_content)
     return retval[0] if retval else None
 
 
-def get_elements_by_class(class_name, html):
-    """Return the content of all tags with the specified class in the passed HTML document as a list"""
+def get_elements_by_class(class_name, html, return_content=True):
+    """Return the content or all tags with the specified class in the passed HTML document as a list"""
     return get_elements_by_attribute(
         'class', r'[^\'"]*\b%s\b[^\'"]*' % re.escape(class_name),
-        html, escape_value=False)
+        html, escape_value=False, return_content=return_content)
 
 
-def get_elements_by_attribute(attribute, value, html, escape_value=True):
-    """Return the content of the tag with the specified attribute in the passed HTML document"""
+def get_elements_by_attribute(attribute, value, html, escape_value=True, return_content=True):
+    """Return the content or the tag with the specified attribute in the passed HTML document"""
 
     value = re.escape(value) if escape_value else value
 
     retlist = []
     for m in re.finditer(r'''(?xs)
-        <([a-zA-Z0-9:._-]+)
+        <(?P<tag>[a-zA-Z0-9:._-]+)
          (?:\s+[a-zA-Z0-9:._-]+(?:=[a-zA-Z0-9:._-]*|="[^"]*"|='[^']*'|))*?
          \s+%s=['"]?%s['"]?
          (?:\s+[a-zA-Z0-9:._-]+(?:=[a-zA-Z0-9:._-]*|="[^"]*"|='[^']*'|))*?
         \s*>
-        (?P<content>.*?)
-        </\1>
     ''' % (re.escape(attribute), value), html):
```

Comment:
The tests are green again, but this regex does not match such an element:
```html
  <div id="player_div_id"
       data-kwik_image="https://www.megatv.com/wp-content/uploads/2021/10/ΠΕΙΡΑΙΑΣ-1024x450.jpg"
       data-share_url="https://www.megatv.com/2021/10/23/egkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia/"
       data-kwik_source="https://c98db5952cb54b358365984178fb898a.msvdn.net/vod/gUgEuW3fjIhP/playlist.m3u8"
       data-kwik_theurl="https%3A%2F%2Fwww.megatv.com%2F2021%2F10%2F23%2Fegkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia&description_url=https%3A%2F%2Fwww.megatv.com%2F2021%2F10%2F23%2Fegkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia&cust_params=Category%3Dnews%2Cautoplay%26SubCategory%3Dmesimvrino-deltio-eidisen%26Item_Id%3D520979%26Tags%3DMEGA%2CMEGA-%CE%93%CE%95%CE%93%CE%9F%CE%9D%CE%9F%CE%A4%CE%91%2C%CE%93%CE%99%CE%91%CE%9D%CE%9D%CE%97%CE%A3-%CE%9C%CE%A9%CE%A1%CE%91%CE%9B%CE%97%CE%A3%2C%CE%98%CE%95%CE%91%CE%A4%CE%A1%CE%9F%2C%CE%A0%CE%95%CE%99%CE%A1%CE%91%CE%99%CE%91%CE%A3"
       data-kwik_start="true"
       data-kwik_category="news"
       data-kwik_label = "Μεσημβρινό Δελτίο Ειδήσεων: Εγκαίνια για τη νέα σκηνή «Ωμέγα» του δημοτικού θεάτρου Πειραιά"
       data-kwik_subs=""></div>
```

So I'm still looking into it.